lfs.h

```diff
@@ -644,7 +655,106 @@ int lfs_fs_traverse(lfs_t *lfs, int (*cb)(void*, lfs_block_t), void *data);
+int lfs_mkdir_ts(lfs_t *lfs, const char *path);
+int lfs_dir_open_ts(lfs_t *lfs, lfs_dir_t *dir, const char *path);
+int lfs_dir_close_ts(lfs_t *lfs, lfs_dir_t *dir);
+int lfs_dir_read_ts(lfs_t *lfs, lfs_dir_t *dir, struct lfs_info *info);
+int lfs_dir_seek_ts(lfs_t *lfs, lfs_dir_t *dir, lfs_off_t off);
+lfs_soff_t lfs_dir_tell_ts(lfs_t *lfs, lfs_dir_t *dir);
+int lfs_dir_rewind_ts(lfs_t *lfs, lfs_dir_t *dir);
+lfs_ssize_t lfs_fs_size_ts(lfs_t *lfs);
+int lfs_fs_traverse_ts(lfs_t *lfs, int (*cb)(void*, lfs_block_t), void *data);
+int lfs_migrate_ts(lfs_t *lfs, const struct lfs_config *cfg);
+
+#define lfs_format lfs_format_ts
+#define lfs_mount lfs_mount_ts
+#define lfs_unmount lfs_unmount_ts
+#define lfs_remove lfs_remove_ts
+#define lfs_rename lfs_rename_ts
+#define lfs_stat lfs_stat_ts
+#define lfs_getattr lfs_getattr_ts
+#define lfs_setattr lfs_setattr_ts
+#define lfs_removeattr lfs_removeattr_ts
+#define lfs_file_open lfs_file_open_ts
+#define lfs_file_opencfg lfs_file_opencfg_ts
+#define lfs_file_close lfs_file_close_ts
+#define lfs_file_sync lfs_file_sync_ts
+#define lfs_file_read lfs_file_read_ts
+#define lfs_file_write lfs_file_write_ts
+#define lfs_file_seek lfs_file_seek_ts
+#define lfs_file_truncate lfs_file_truncate_ts
+#define lfs_file_tell lfs_file_tell_ts
+#define lfs_file_rewind lfs_file_rewind_ts
+#define lfs_file_size lfs_file_size_ts
+#define lfs_mkdir lfs_mkdir_ts
+#define lfs_dir_open lfs_dir_open_ts
+#define lfs_dir_close lfs_dir_close_ts
+#define lfs_dir_read lfs_dir_read_ts
+#define lfs_dir_seek lfs_dir_seek_ts
+#define lfs_dir_tell lfs_dir_tell_ts
+#define lfs_dir_rewind lfs_dir_rewind_ts
+#define lfs_fs_size lfs_fs_size_ts
+#define lfs_fs_traverse lfs_fs_traverse_ts
+#define lfs_migrate lfs_migrate_ts
+
+#else
+
+#define lfs_format lfs_format_raw
+#define lfs_mount lfs_mount_raw
+#define lfs_unmount lfs_unmount_raw
+#define lfs_remove lfs_remove_raw
+#define lfs_rename lfs_rename_raw
+#define lfs_stat lfs_stat_raw
+#define lfs_getattr lfs_getattr_raw
+#define lfs_setattr lfs_setattr_raw
+#define lfs_removeattr lfs_removeattr_raw
+#define lfs_file_open lfs_file_open_raw
+#define lfs_file_opencfg lfs_file_opencfg_raw
+#define lfs_file_close lfs_file_close_raw
+#define lfs_file_sync lfs_file_sync_raw
+#define lfs_file_read lfs_file_read_raw
+#define lfs_file_write lfs_file_write_raw
+#define lfs_file_seek lfs_file_seek_raw
+#define lfs_file_truncate lfs_file_truncate_raw
+#define lfs_file_tell lfs_file_tell_raw
+#define lfs_file_rewind lfs_file_rewind_raw
+#define lfs_file_size lfs_file_size_raw
+#define lfs_mkdir lfs_mkdir_raw
+#define lfs_dir_open lfs_dir_open_raw
+#define lfs_dir_close lfs_dir_close_raw
+#define lfs_dir_read lfs_dir_read_raw
+#define lfs_dir_seek lfs_dir_seek_raw
+#define lfs_dir_tell lfs_dir_tell_raw
+#define lfs_dir_rewind lfs_dir_rewind_raw
+#define lfs_fs_size lfs_fs_size_raw
+#define lfs_fs_traverse lfs_fs_traverse_raw
+#define lfs_migrate lfs_migrate_raw
```

Comment:
I'm not a fan of redefining functions, as they can cause problems for debuggers. It may be tedious, but I believe each of these should get its own wrapper:
``` c
int lfs_file_close(lfs_t *lfs, lfs_file_t *file) {
    return lfs_file_close_raw(lfs, file);
}
```

Thinking ahead, I'm planning to have logging overhead in each of these, so I don't think any will end up as a simple passthrough. It should be easy enough for the compiler to inline these anyways.

Reply:
fixed